streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -87,59 +85,64 @@ public void before() {
+
+        Assert.assertEquals(globalStreams.state(), KafkaStreams.State.CREATED);
         Assert.assertEquals(stateListener.numChanges, 0);
 
-        streams.start();
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                return streams.state() == KafkaStreams.State.RUNNING;
-            }
-        }, 10 * 1000, "Streams never started.");
-        streams.close();
-        Assert.assertEquals(streams.state(), KafkaStreams.State.NOT_RUNNING);
+        globalStreams.start();
+        TestUtils.waitForCondition(
+            () -> globalStreams.state() == KafkaStreams.State.RUNNING,
+            10 * 1000,
+            "Streams never started.");
+
+        globalStreams.close();
+
+        Assert.assertEquals(globalStreams.state(), KafkaStreams.State.NOT_RUNNING);
     }
 
     @Test
     public void testStateCloseAfterCreate() {
-        final StreamsBuilder builder = new StreamsBuilder();
         final KafkaStreams streams = new KafkaStreams(builder.build(), props);
 
-        final StateListenerStub stateListener = new StateListenerStub();
-        streams.setStateListener(stateListener);
-        streams.close();
+        try {
+            final StateListenerStub stateListener = new StateListenerStub();
+            streams.setStateListener(stateListener);
+        } finally {
+            streams.close();
+        }
+
         Assert.assertEquals(KafkaStreams.State.NOT_RUNNING, streams.state());
     }
 
     @Test
     public void shouldCleanupResourcesOnCloseWithoutPreviousStart() throws Exception {
-        final StreamsBuilder builder = new StreamsBuilder();
         builder.globalTable("anyTopic");
         final List<Node> nodes = Arrays.asList(new Node(0, "localhost", 8121));
         final Cluster cluster = new Cluster("mockClusterId", nodes,
-                                            Collections.<PartitionInfo>emptySet(), Collections.<String>emptySet(),
-                                            Collections.<String>emptySet(), nodes.get(0));
+                                            Collections.emptySet(), Collections.<String>emptySet(),
+                                            Collections.emptySet(), nodes.get(0));
         final MockClientSupplier clientSupplier = new MockClientSupplier();
         clientSupplier.setClusterForAdminClient(cluster);
         final KafkaStreams streams = new KafkaStreams(builder.build(), props, clientSupplier);
         streams.close();
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                return streams.state() == KafkaStreams.State.NOT_RUNNING;
-            }
-        }, 10 * 1000, "Streams never stopped.");
+        TestUtils.waitForCondition(
+            () -> streams.state() == KafkaStreams.State.NOT_RUNNING,
+            10 * 1000,
+            "Streams never stopped.");
```

Comment:
Just out of curiosity, how does this differ from the following?
```
assertTrue(streams.close(10, TimeUnit.SECONDS));
```

Reply:
Not sure... Seems we could rewrite it. \cc @guozhangwang @bbejeck WDYT?